napari/layers/base/base.py

```diff
@@ -1594,9 +1595,36 @@ def _update_draw(
             corners[:, displayed_axes] = data_bbox_clipped
             self.corner_pixels = corners
 
+    def _get_source_str(self):
```

Comment:
Not sure what others think but maybe this should be a layer util and take a layer rather than being a method on `base`?